### PR TITLE
Fix ComQueue Does Not Deallocate Buffers on Overflow

### PR DESCRIPTION
**Pull request by DJKessler**
| | |
|:---|:---|
|**_Related Issue(s)_**|  #2066  |
|**_Has Unit Tests (y/n)_**| y |
|**_Documentation Included (y/n)_**| y |

---
## Change Description

Fix potential memory leak when a sent `Fw::Buffer` overflows `ComQueue`.

* Added `Fw.BufferSend` `deallocate` port to `ComQueue`
* Updated FPP to v2.2.0a2
* Change `buffQueueIn` port from `drop` to `hook`
* Added queue overflow hook method implementation
* Added return status to `ComQueue::enqueue()`
* `Fw::Buffer` is now deallocated on queue overflow
* Enabled `UT_AUTO_HELPERS` in `ComQueue` UT build
* Updated `ComQueue` UTs

### CODE REVIEW

File: Svc/ComQueue/ComQueue.cpp [hunk 1]
```diff
@@ -139,7 +139,11 @@ void ComQueue::buffQueueIn_handler(const NATIVE_INT_TYPE portNum, Fw::Buffer& fw
     // Ensure that the port number of buffQueueIn is consistent with the expectation
     FW_ASSERT(portNum >= 0 && portNum < BUFFER_PORT_COUNT, portNum);
     FW_ASSERT(queueNum < TOTAL_PORT_COUNT);
-    this->enqueue(queueNum, QueueType::BUFFER_QUEUE, reinterpret_cast<const U8*>(&fwBuffer), sizeof(Fw::Buffer));
+    bool status =
+        this->enqueue(queueNum, QueueType::BUFFER_QUEUE, reinterpret_cast<const U8*>(&fwBuffer), sizeof(Fw::Buffer));
+    if (!status) {
+        this->buffQueueIn_overflowHook(portNum, fwBuffer);
```
Comment (LeStarch) [inline Svc/ComQueue/ComQueue.cpp]: Perhaps we can just call this->deallocate_out?

File: Svc/ComQueue/ComQueue.hpp [hunk 2]
```diff
@@ -151,13 +151,23 @@ class ComQueue : public ComQueueComponentBase {
...
     );
 
+    // ----------------------------------------------------------------------
+    // Hook implementations for typed async input ports
+    // ----------------------------------------------------------------------
+
+    //! Queue overflow hook method that deallocates the fwBuffer
+    //!
+    void buffQueueIn_overflowHook(FwIndexType portNum, /*!< The port number*/
+                                  Fw::Buffer& fwBuffer /*!< The buffer*/
```
Comment (LeStarch) [inline Svc/ComQueue/ComQueue.hpp]: The new style is:

```
//!< The port number
```

**Review by LeStarch — CHANGES_REQUESTED** [review]
There are several things that need to be fixed:

1. In the com queue function, we should explicitly discard the new enqueue status with `(void) this->enqueue...`
2. I don't see any tests proving that the drop function is called on overflow.  Can we add those.
3. CI needs to pass

File: Svc/ComQueue/ComQueue.cpp (hunk 1 above)
Comment (DJKessler) [reply Svc/ComQueue/ComQueue.cpp]: Fixed in 658d2a43e.

File: Svc/ComQueue/ComQueue.hpp (hunk 2 above)
Comment (DJKessler) [reply Svc/ComQueue/ComQueue.hpp]: Fixed in c2e203c7d.

**Review by LeStarch — APPROVED** [review]

**Comment by LeStarch** [comment]
Skipping known ci failure.  Nice work!

**Pull request merged**